### PR TITLE
Document release process

### LINKED ISSUE #222: Document release process
The current release process is undocumented, especially how to generate the new changelog.

@tomasbedrich Would you be up to document the corresponding steps?

### PR DESCRIPTION
**Pull request by tomasbedrich**
Fixes #222

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -109,3 +109,15 @@ Coding style
   third-party module, please consult it on GitHub before.
 - `Please use regular expressions only as a last resort. <http://imgur.com/j3G9xyP>`_ When possible, use string manipulations,
   such as :code:`split()` and then list operations. It is more readable.
+
+
+Release process
+-------------------------------------------------------------------------------
+
+1. Pick a suitable semantic version number. We adhere to `generic rules <https://docs.npmjs.com/about-semantic-versioning#incrementing-semantic-versions-in-published-packages>`_ with an exception of our `specific deprecation policy <https://github.com/tomasbedrich/pycaching/blob/master/docs/api.rst?plain=1#L7-L11>`_.
```
Comment (tomasbedrich) [inline CONTRIBUTING.rst:117]: @FriedrichFroebel what are your thoughts about the deprecation policy? I am wondering if we should update it to match semver, i.e. for deprecated methods to be removed together with major version bump.

Comment (FriedrichFroebel) [reply CONTRIBUTING.rst:117]: In my personal opinion, the current deprecation policy is fine and apparently no one complained about it in the past (although deprecations and releases generally are rather sparse). I would reserve major version changes for really breaking changes like dropping old Python versions or old dependency versions, which we do even less.

**Review by FriedrichFroebel — APPROVED** [review]

**Comment by FriedrichFroebel** [comment]
@tomasbedrich As the `master` branch is protected, I cannot push to it directly and will have to create a PR which AFAIK usually requires approval from you (or one of the other (currently inactive) collaborators). I do not know if there is a clean solution to this for now, but to clarify this, I did not change the version number for now.

**Comment by FriedrichFroebel** [comment]
I have added a corresponding PR. Yes, the goal is to fulfill the request from #218 while checking that everything works as intended.

**Comment by FriedrichFroebel** [comment]
Thanks, the remaining steps seem to have worked flawlessly.

**Pull request merged**